newIDE/app/src/EventsSheet/index.js

```diff
@@ -235,11 +252,25 @@ export default class EventsSheet extends React.Component<Props, State> {
     fontSize: 14,
   };
 
+  constructor(props: ComponentProps) {
+    super(props);
+    this.addNewEvent = addPostBadgePreHookIfLocked(
+      this.props.authenticatedUser,
+      'trivial_first-event',
```

Comment:
should those magic strings be in a list of constants somewhere? To avoid typing mistakes.

I'd recommend a new file `src/Utils/GDevelopServices/Achievement.js` or `Badge.js` -> you can add the list here, plus the typings, plus the api calls